File: src/main/java/kr/hyfata/zero/ZeroCore.java
package kr.hyfata.zero;

import kr.hyfata.zero.gui.InventoryEventListener;
import kr.hyfata.zero.util.config.ConfigUtil;
import org.bukkit.plugin.java.JavaPlugin;

public final class ZeroCore extends JavaPlugin {
    private static ZeroConfig configModules;
    private ZeroModules modules;

    @Override
    public void onEnable() {
        getServer().getPluginManager().registerEvents(new InventoryEventListener(), this);
        ConfigUtil.init(this);
        initModules();
    }

    @Override
    public void onDisable() {
        configModules.save();
        modules.onDisable();
    }

    private void initModules() {
        configModules = new ZeroConfig(this);
        modules = new ZeroModules(this);
    }

    public static ZeroConfig getZeroConfig() {
        return configModules;
    }
}

File: src/main/java/kr/hyfata/zero/ZeroCore.kt
package kr.hyfata.zero

import kr.hyfata.zero.gui.InventoryEventListener
import org.bukkit.plugin.java.JavaPlugin

class ZeroCore : JavaPlugin() {
    private var modules: ZeroModules? = null

    override fun onEnable() {
        server.pluginManager.registerEvents(InventoryEventListener(), this)
        server.pluginManager.registerEvents(ZeroGlobalListener(), this)
        initModules()
    }

    override fun onDisable() {
        configModules!!.save()
        modules!!.onDisable()
    }

    private fun initModules() {
        configModules = ZeroConfig(this)
        modules = ZeroModules(this)
    }

    companion object {
        private var configModules: ZeroConfig? = null
        val zeroConfig: ZeroConfig
            get() = configModules!!
    }
}
